src/ui/FieldValue/FieldValue.ts

```diff
@@ -399,35 +406,64 @@ export class FieldValue extends Component {
       return componentIsAStandardFacet;
     });
 
-    const atLeastOneFacetIsEnabled = facets.length > 0;
-
-    if (atLeastOneFacetIsEnabled) {
-      const selected = find<Facet>(facets, (facet: Facet) => {
+    if (facets.length) {
+      const isValueSelected = !!find<Facet>(facets, facet => {
         const facetValue = facet.values.get(originalFacetValue);
         return facetValue && facetValue.selected;
       });
 
-      const label = selected ? l('RemoveFilterOn', renderedFacetValue) : l('FilterOn', renderedFacetValue);
-      new AccessibleButton()
-        .withTitle(label)
-        .withLabel(label)
-        .withElement(element)
-        .withSelectAction(() => this.handleSelection(selected, facets, originalFacetValue))
-        .build();
+      const selectAction = () => this.handleFacetSelection(isValueSelected, facets, originalFacetValue);
+      this.buildClickableElement(element, !!isValueSelected, renderedFacetValue, selectAction);
+    }
+  }
+
+  private bindDynamicFacets(element: HTMLElement, originalFacetValue: string, renderedFacetValue: string) {
+    const dynamicFacetAttributeName = QueryStateModel.getDynamicFacetId(this.options.facet.replace(/^@/, ''));
+    const dynamicFacets = filter<DynamicFacet>(
+      this.componentStateModel.get(dynamicFacetAttributeName),
+      (component: Component) => component.type === DynamicFacet.ID && !component.disabled
+    );
 
-      if (selected) {
-        $$(element).addClass('coveo-selected');
-      }
-      $$(element).addClass('coveo-clickable');
+    if (dynamicFacets.length) {
+      const isValueSelected = !!find<DynamicFacet>(dynamicFacets, dynamicFacet => dynamicFacet.values.hasSelectedValue(originalFacetValue));
+
+      const selectAction = () => this.handleDynamicFacetSelection(isValueSelected, dynamicFacets, originalFacetValue);
+      this.buildClickableElement(element, isValueSelected, renderedFacetValue, selectAction);
     }
   }
 
-  private handleSelection(selected: Facet, facets: Facet[], value: string) {
-    if (selected != null) {
-      each(facets, (facet: Facet) => facet.deselectValue(value));
-    } else {
-      each(facets, (facet: Facet) => facet.selectValue(value));
+  private buildClickableElement(element: HTMLElement, isSelected: boolean, value: string, selectAction: () => void) {
+    const label = isSelected ? l('RemoveFilterOn', value) : l('FilterOn', value);
+    new AccessibleButton()
+      .withTitle(label)
+      .withLabel(label)
+      .withElement(element)
+      .withSelectAction(selectAction)
+      .build();
+
+    if (isSelected) {
+      $$(element).addClass('coveo-selected');
     }
+    $$(element).addClass('coveo-clickable');
+  }
+
+  private handleFacetSelection(isValueSelected: boolean, facets: Facet[], value: string) {
```

Comment:
Is there a common interface between `Facet` and `DynamicFacet`? I think it is okay to have separate methods for  `handleFacetSelection` and `handleDynamicFacetSelection`, but since they are the same except for the type of the second arg, it would be interesting if we could reuse the same method.

Reply:
Can't do that because they have different call signature (was going for it at first)